src/main/java/org/apache/datasketches/kll/KllDoublesSketch.java

```diff
@@ -287,32 +74,22 @@ public KllDoublesSketch(final int k) {
    * @param m minimum level size. Default is 8.
    */
   private KllDoublesSketch(final int k, final int m, final boolean compatible) {
-    checkK(k);
-    k_ = k;
-    minK_ = k;
-    m_ = m;
-    numLevels_ = 1;
-    levels_ = new int[] {k, k};
+    super(k, m, compatible);
     items_ = new double[k];
     minValue_ = Double.NaN;
     maxValue_ = Double.NaN;
-    isLevelZeroSorted_ = false;
-    this.compatible = compatible;
-
   }
 
   /**
-   * Off-heap constructor.
+   * Private heapify constructor.
    * @param mem Memory object that contains data serialized by this sketch.
    */
   private KllDoublesSketch(final Memory mem) {
-    m_ = DEFAULT_M;
-    k_ = mem.getShort(K_SHORT) & 0xffff;
+    super(mem.getShort(K_SHORT) & 0xffff, DEFAULT_M, true);
```

Comment:
I don't understand where the preamble is handled here, is it handled in the public heapify method first?

Reply:
The public static heapify does all the checking of the PreambleInts, SerVer, FamID and Flags.  Once these are determined to be correct, the private constructor, KllDoublesSketch(mem) does the work of populating the sketch on the heap.  The sketch in memory (heap) doesn't care about the PreambleInts, SerVer, FamID fields as those are only used to validate the serialized image, which can be stored anywhere (disk, off-heap, etc.).  

The key non-static variables of the sketch are initialized by the BaseKllSketch constructor, and then changed as necessary by the remainder of the private KllDoublesSketch(mem) construtor.

Reply:
This design makes a lot of sense.